url054c08669f_jules2689_CSI4124_tgz-pJ8-SMThemeParkExperi3J8/src/simModel/RVPs.java

```diff
@@ -0,0 +1,243 @@
+		interArrDistGI = new Exponential(1.0 / MEAN_ARRL_GI_1, new MersenneTwister(sd.arrGI));
+		interArrDistRC = new Exponential(1.0 / MEAN_ARRL_RC_1,new MersenneTwister(sd.arrRC));
+	}
+
+	//
+	protected double uCustomersFP() {
+		double mean = getMeanTimeFP();
+		double nxtArrival = model.getClock() + interArrDistFP.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersSH() {
+		double mean = getMeanTimeSH();
+		double nxtArrival = model.getClock() + interArrDistSH.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersGI() {
+		double mean = getMeanTimeGI();
+		double nxtArrival = model.getClock() + interArrDistGI.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersRC() {
+		double mean = getMeanTimeRC();
+		double nxtArrival = model.getClock() + interArrDistRC.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	//TODO: Add to Documentation
+	protected double getMeanTimeFP() {
+		double mean;
+
+		if (model.getClock() < 60)
+			mean = MEAN_ARRL_FP_1;
+		else if (model.getClock() < 120)
+			mean = MEAN_ARRL_FP_2;
+		else if (model.getClock() < 180)
+			mean = MEAN_ARRL_FP_3;
+		else if (model.getClock() < 240)
+			mean = MEAN_ARRL_FP_4;
+		else if (model.getClock() < 300)
+			mean = MEAN_ARRL_FP_5;
+		else if (model.getClock() < 360)
+			mean = MEAN_ARRL_FP_6;
+		else if (model.getClock() < 420)
+			mean = MEAN_ARRL_FP_7;
+		else if (model.getClock() < 480)
+			mean = MEAN_ARRL_FP_8;
+		else if (model.getClock() < 540)
+			mean = MEAN_ARRL_FP_9;
+		else if (model.getClock() < 600)
+			mean = MEAN_ARRL_FP_10;
+		else if (model.getClock() < 660)
+			mean = MEAN_ARRL_FP_11;
+		else
+			mean = MEAN_ARRL_FP_12;
+
+		return mean;
+	}
+	
+	protected double getMeanTimeSH() {
+		double mean;
+
+		if (model.getClock() < 60)
+			mean = MEAN_ARRL_SH_1;
+		else if (model.getClock() < 120)
+			mean = MEAN_ARRL_SH_2;
+		else if (model.getClock() < 180)
+			mean = MEAN_ARRL_SH_3;
+		else if (model.getClock() < 240)
+			mean = MEAN_ARRL_SH_4;
+		else if (model.getClock() < 300)
+			mean = MEAN_ARRL_SH_5;
+		else if (model.getClock() < 360)
+			mean = MEAN_ARRL_SH_6;
+		else if (model.getClock() < 420)
+			mean = MEAN_ARRL_SH_7;
+		else if (model.getClock() < 480)
+			mean = MEAN_ARRL_SH_8;
+		else if (model.getClock() < 540)
+			mean = MEAN_ARRL_SH_9;
+		else if (model.getClock() < 600)
+			mean = MEAN_ARRL_SH_10;
+		else if (model.getClock() < 660)
+			mean = MEAN_ARRL_SH_11;
+		else
+			mean = MEAN_ARRL_SH_12;
+
+		return mean;
+	}
+	
+	protected double getMeanTimeGI() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.